Remove package-lock.json and add it to gitignore

coyotte508 opened the pull request:
package-lock.json was not updated since 2.0.0

It makes sense to only keep one. Here's a stackoverflow [thread](https://stackoverflow.com/questions/44552348/should-i-commit-yarn-lock-and-package-lock-json-files) with a debate on the subject.

The lockfile is ignored when the project is used as a library, so removing package-lock.json does not change anything for users of the library.

omitobi commented:
I think I'll argue this proposal.
A Package needs to lock its dependencies at the tested and working versions across the machine/packages where it is used.
Referring to that thread, I could even conclude that its safer to leave it than remove it. If `package-lock.json` is ignored when a codebase is an npm package, then leaving it there would do no harm.

My proposal instead is if there's a need to update the libraries this package depends on, then changes should be made on `package.json` file, the changes tested, and then a new version of the package released.

coyotte508 commented:
The point is that right now, yarn is used, and yarn.lock is the reference. I didn't remove `yarn.lock` in the commit, only `package-lock.json`.

`package-lock.json` is outdated, and anyone using it would be using the wrong lockfile. It can confuse users. 

Of course both could be kept and updated but is it worth it? It could create conflicts if both `npm install` & `yarn install` are run in the same folder with different versions of dependencies in their lockfiles. The maintainer would have to edit `package-lock.json` and `yarn.lock` manually to make sure they stay the same, which is hell.

Take webpack, a hugely depended package: https://github.com/webpack/webpack. They only have `yarn.lock` and added `package-lock.json` to their gitignore: https://github.com/webpack/webpack/blob/master/.gitignore.

Taking the first two library packages I could think of: [lodash](https://github.com/lodash/lodash) only has `package-lock.json`, and [express](https://github.com/expressjs/express) has neither.
